src/basic/unit-name.c

```diff
@@ -241,25 +241,45 @@ int unit_name_change_suffix(const char *n, const char *suffix, char **ret) {
 }
 
 int unit_name_build(const char *prefix, const char *instance, const char *suffix, char **ret) {
-        char *s;
+        UnitType type;
 
         assert(prefix);
         assert(suffix);
         assert(ret);
 
+        if (isempty(suffix))
+                return -EINVAL;
+        if (suffix[0] != '.')
+                return -EINVAL;
```

Comment:
The first check is actually a subset of the second...

And shouldn't this be an `assert` instead?

Reply:
true, the first check is redundant indeed...

I think it's a good idea to place assert()s to catch programming errors while catching invalid data errors with runtime errors. I'd claim that a parameter being NULL where it shouldn't is really a programming error, while looking into a string could be an input data error, and hence should be handled more defensively...

Reply:
I prepped a short patch for this locally. Let's discuss it in the new PR.